simexpal/build.py

```diff
@@ -264,12 +264,27 @@ def git_ref_changed():
 						build.clone_dir,
 						generic_tag])
 			else:
-				# Recreate the source directory
-				util.try_rmtree(build.source_dir)
-				util.try_mkdir(build.source_dir)
+				branch = build.revision.version_for_build(build.name)
+				if os.path.isdir(build.source_dir):
+					# Repository already cloned, check for local changes
+					status = subprocess.check_output(['git', 'status', '--porcelain'], cwd=build.source_dir)
```

Comment:
What you need to check here is: am I still on the branch given by the config? If not, and there are local changes: print error message, and do not proceed. If there are no local changes, checkout to configured branch. Now is the question: are there any local commits not pushed? If so: print error message, do not proceed. If everything is fine: pull in the latest changes from origin.

Reply:
The logic is now as follows:

1. if the repository has not been cloned yet:
    1. run `git clone <repo_url>`
    2. run `git checkout <branch_name>`, where `branch_name` is the `build_version` specified in the `experiments.yml`
2. if the repository has already been cloned:
    1. if there are local changes using (checked using `git status --porcelain`) or unpushed commits (checked using `git log --branches --not --remotes`): warn the user of local changes and do not proceed
    2. if there are no local changes and no unpushed commits: checkout given branch by running `git checkout <branch_name>` and then `git pull`

If the `--reclone` flag was set, the installed repository is deleted and recloned, meaning that all the following build phases are executed, as well.